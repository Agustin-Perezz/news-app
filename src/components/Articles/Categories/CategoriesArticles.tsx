import { Box, Chip } from '@mui/material';
import React from 'react';
import { CategoriesProps } from '../../../types';

interface Props {
  listCategories: CategoriesProps[];
}

export const CategoriesArticles: React.FC<Props> = ({ listCategories }) => {
  return (
    <Box sx={{ justifyContent: 'flex-start', my: '2' }}>
      {listCategories.map((categori) => (
        <Chip
          label={categori.type}
          key={categori.color}
          sx={{
            position: 'absolute',
            top: '8px',
            right: '5px',
            color: categori.color,
            backgroundColor: '#0A1929',
            height: '29px',
            border: `1px solid ${categori.color}`,
          }}
        />
      ))}
    </Box>
  );
};
