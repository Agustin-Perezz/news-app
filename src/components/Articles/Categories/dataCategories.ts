export const dataCategories = [
  {
    type: 'general',
    color: '#27998D ',
  },
  {
    type: 'science',
    color: '#FFC300 ',
  },
  {
    type: 'sports',
    color: '#3572DC ',
  },
  {
    type: 'business',
    color: '#A4A72D',
  },
  {
    type: 'health',
    color: '#2CBE65 ',
  },
  {
    type: 'entertainment',
    color: '#CD3C3C',
  },
  {
    type: 'tech',
    color: '#9D2672 ',
  },
  {
    type: 'pilitics',
    color: '#269D98 ',
  },
  {
    type: 'food',
    color: '#767E17 ',
  },
  {
    type: 'travel',
    color: '#2F7DB9 ',
  },
];
