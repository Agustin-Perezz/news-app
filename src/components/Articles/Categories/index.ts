export * from './CategoriesArticles';
export * from './dataCategories';
export * from './Categori';
