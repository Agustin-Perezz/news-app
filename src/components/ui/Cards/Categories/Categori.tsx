import { Chip } from '@mui/material';
import { CategoriesProps } from '../../../../types';

interface Props {
  categori: CategoriesProps;
}

export const Categori: React.FC<Props> = ({ categori }) => {
  return (
    <Chip
      label={categori.type}
      sx={{
        color: categori.color,
        backgroundColor: '#0A1929',
        height: '25px',
        border: `1px solid ${categori.color}`,
        marginRight: '5px',
        marginBottom: '5px',
      }}
    />
  );
};
