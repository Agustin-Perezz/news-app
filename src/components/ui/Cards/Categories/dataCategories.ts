export const categoryColorsAccordingToType: Record<string, string> = {
  general: '#27998D',
  science: '#FFC300',
  sports: '#3572DC',
  business: '#A4A72D',
  health: '#2CBE65',
  entertainment: '#CD3C3C',
  tech: '#9D2672',
  politics: '#269D98',
  food: '#767E17',
  travel: '#2F7DB9',
};
