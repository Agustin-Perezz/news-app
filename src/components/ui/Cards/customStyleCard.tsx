import { styled, createTheme } from '@mui/material/styles';
import MuiCard from '@mui/material/Card';
import MuiChip from '@mui/material/Chip';

export const tabFont = createTheme({
  typography: {
    fontFamily: ['Heebo', 'sans-serif'].join(','),
  },
  palette: { secondary: { main: '#2475C7' } },
});

export const CustomCard = styled(MuiCard)(() => ({
  '&:hover': {
    boxShadow: '0px 2px 18px 0px rgba(0,0,0,0.8)',
    transform: 'scale(1.02)',
    transition: 'all .2s ease-in-out',
    cursor: 'pointer',
  },
  maxHeight: 'auto',
  minHeight: '447px',
  width: '278px',
}));

export const CustomChip = styled(MuiChip)(() => ({
  position: 'absolute',
  top: '8px',
  right: '5px',
  color: 'ghostwhite',
  backgroundColor: '#0A1929',
  opacity: '0.6',
  height: '29px',
  border: '1px solid #9393A5',
}));
