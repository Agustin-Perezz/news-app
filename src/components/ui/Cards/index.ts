export * from './ListCards';
export * from './customStyleCard';
