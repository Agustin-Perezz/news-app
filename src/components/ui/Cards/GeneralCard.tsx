import { CardActionArea, CardContent, Typography, Box } from '@mui/material';
import { NewsProps } from '../../../types/news-props';
import { CategoriesArticles } from './Categories';
import { CustomCard, CustomChip } from './CustomStylesCard';
import { useLoadImage } from '../../../hooks';

import load_image from '../../../assets/images/load-image.jpg';

interface Props {
  dataArticle: NewsProps;
}

export const GeneralCard: React.FC<Props> = ({ dataArticle }) => {
  const { loaded, onError, onLoad } = useLoadImage();

  return (
    <CustomCard>
      <CardActionArea target="_blank" href={dataArticle.url} sx={{ height: '470px' }}>
        <Box sx={{ position: 'relative' }}>
          <img
            style={{ height: '185px' }}
            onLoad={onLoad}
            alt={dataArticle.nameSource}
            src={loaded ? dataArticle.image_url : load_image}
            onError={({ currentTarget }) => onError({ currentTarget })}
          />
          <CustomChip label={dataArticle.published_at} />
        </Box>
        <CardContent>
          <Typography data-testid="card-title" variant="subtitle1">
            {dataArticle.title}
          </Typography>
          <Typography data-testid="card-description" variant="body2" color="text.secondary">
            {dataArticle.description}
          </Typography>
          {dataArticle.categories && <CategoriesArticles listCategories={dataArticle.categories} />}
          {dataArticle.nameSource && (
            <Box sx={{ display: 'flex', marginTop: '10px' }}>
              <Typography variant="subtitle2" mr={1}>
                Source:
              </Typography>
              <Typography variant="body2" color="text.secondary" sx={{ lineHeight: '1.57' }}>
                {dataArticle.nameSource}
              </Typography>
            </Box>
          )}
        </CardContent>
      </CardActionArea>
    </CustomCard>
  );
};
