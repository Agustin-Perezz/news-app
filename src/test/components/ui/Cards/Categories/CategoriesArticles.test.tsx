import { render, screen } from '@testing-library/react';
import { CategoriesArticles } from '../../../../../components/ui/Cards/Categories';
import { listCategories } from './categoriesMock';

describe('Test in <CategoriesArticles/>', () => {
  it('should match snapshot', () => {
    const { container } = render(<CategoriesArticles listCategories={listCategories} />);
    expect(container).toMatchSnapshot();
  });
  it('should show title categories', () => {
    render(<CategoriesArticles listCategories={listCategories} />);
    screen.getByText('science');
    screen.getByText('general');
  });
});
