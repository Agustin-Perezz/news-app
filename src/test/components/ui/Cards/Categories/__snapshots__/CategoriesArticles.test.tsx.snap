// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <CategoriesArticles/> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-3yoo1r"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-1ij1w03-MuiTypography-root"
    >
      Categories:
    </h6>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1u2dp2a-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        general
      </span>
    </div>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1qzrz4y-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        science
      </span>
    </div>
  </div>
</div>
`;
